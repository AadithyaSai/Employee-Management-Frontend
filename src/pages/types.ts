export interface JWTPayload {
  email: string;
  id: number;
  role: string;
}
