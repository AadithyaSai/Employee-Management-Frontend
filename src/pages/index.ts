import CreateEmployee from "./createEmployee/createEmployee";
import Login from "./login/Login";
import MainPage from "./mainPage/mainPage";

export { CreateEmployee, Login, MainPage };
