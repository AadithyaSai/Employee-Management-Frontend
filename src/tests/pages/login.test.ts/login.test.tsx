import { describe, it, expect, beforeEach, vi } from "vitest";
import { render, screen, waitFor } from "@testing-library/react";
import { Provider } from "react-redux";
import { configureStore } from "@reduxjs/toolkit";
import userEvent from "@testing-library/user-event";
import { Login } from "../../../pages";
import "@testing-library/jest-dom";

// Mock react-router-dom's useNavigate
const mockNavigate = vi.fn();
vi.mock("react-router-dom", async () => {
  const actual = await vi.importActual("react-router-dom");
  return {
    ...actual,
    useNavigate: () => mockNavigate,
  };
});
// // Mock the store
const mockStore = configureStore({
  reducer: {
    employee: (state = []) => state,
    department: (state = []) => state,
    api: (state = {}) => state,
  },
});
// Mock the login mutation
const mockLoginMutation = vi.fn();
vi.mock("../../../api-service/auth/login.api", () => ({
  useLoginMutation: () => [
    () => ({
      unwrap: () => mockLoginMutation(),
    }),
    { isLoading: false },
  ],
}));

describe("Login Page", () => {
  const renderLogin = () => {
    return render(
      <Provider store={mockStore}>
        <Login />
      </Provider>
    );
  };

  beforeEach(() => {
    vi.clearAllMocks();
  });

  it("should match snapshot", () => {
    const { container } = renderLogin();
    expect(container).toMatchSnapshot();
  });

  // Test 2: Render login form with all required elements
  it("should render login form with all required elements", () => {
    renderLogin();

    // Check if username input is present
    const usernameInput = screen.getByLabelText("Username");
    expect(usernameInput).toBeInTheDocument();

    // Check if password input is present
    const passwordInput = screen.getByLabelText("Password");
    expect(passwordInput).toBeInTheDocument();

    // Check if submit button is present
    const submitButton = screen.getByRole("button", { name: /log in/i });
    expect(submitButton).toBeInTheDocument();
    expect(submitButton).toHaveAttribute("type", "button");
  });

  it("should show error when password is too short", async () => {
    renderLogin();
    const passwordInput = screen.getByLabelText("Password");
    // Type a username longer than 30 characters
    await userEvent.type(passwordInput, "a".repeat(7));
    // Check if error message is displayed
    expect(screen.getByText("Password is too short")).toBeInTheDocument();
  });

  it("should handle successful login", async () => {
    mockLoginMutation.mockResolvedValueOnce({ token: "fake-token" });

    renderLogin();

    // Fill in the form
    const usernameInput = screen.getByLabelText("Username");
    const passwordInput = screen.getByLabelText("Password");
    const submitButton = screen.getByRole("button", { name: /log in/i });

    await userEvent.type(usernameInput, "test@example.com");
    await userEvent.type(passwordInput, "password123");

    // Submit the form
    await userEvent.click(submitButton);

    // Wait for the login mutation to resolve and navigation to occur
    await waitFor(() => {
      expect(mockLoginMutation).toHaveBeenCalled();
      expect(mockNavigate).toHaveBeenCalledWith("/employees");
    });
  });
});
