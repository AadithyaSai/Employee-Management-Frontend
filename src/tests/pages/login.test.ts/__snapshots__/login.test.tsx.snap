// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Login Page > should match snapshot 1`] = `
<div>
  <div
    class="loader-bg disappear"
  >
    <div
      class="loader-container"
    >
      <svg
        viewBox="0 0 200 200"
        xmlns="http://www.w3.org/2000/svg"
      >
        <lineargradient
          id="a11"
        >
          <stop
            offset="0"
            stop-color="#03AEEE"
            stop-opacity="0"
          />
          <stop
            offset="1"
            stop-color="#03AEEE"
          />
        </lineargradient>
        <circle
          cx="100"
          cy="100"
          fill="none"
          r="70"
          stroke="url(#a11)"
          stroke-dasharray="0 44 0 44 0 44 0 44 0 360"
          stroke-linecap="round"
          stroke-width="15"
          transform-origin="center"
        />
      </svg>
    </div>
  </div>
  <div
    class="login-content"
  >
    <aside
      class="login-aside"
    >
      <img
        alt="stock image for login"
        class="circle-img"
        src="/assets/kv-login.jpeg"
      />
    </aside>
    <main
      class="login-main"
    >
      <div
        class="login-form-content"
      >
        <div
          class="logo-div"
        >
          <a
            href="#"
          >
            <img
              alt="logo"
              class="logo"
              src="/assets/kv-logo.png"
            />
          </a>
        </div>
        <div>
          <div
            class="text-input-field-div"
          >
            <label
              class="label--animated"
            >
              <input
                class="input--animated"
                name="username"
                placeholder=""
                required=""
                type="text"
                value=""
              />
              <span>
                Username
              </span>
              <div
                class="end-adornment"
              >
                <button
                  style="background: none; font-size: 16px; cursor: pointer; display: none;"
                  type="button"
                >
                  Ã—
                </button>
              </div>
            </label>
          </div>
          <p
            style="margin-left: 6px; color: red;"
          >
            Username is required
          </p>
        </div>
        <div>
          <div
            style="margin-bottom: 12px;"
          >
            <div
              class="text-input-field-div"
            >
              <label
                class="label--animated"
              >
                <input
                  class="input--animated"
                  name="password"
                  placeholder=""
                  required=""
                  type="password"
                  value=""
                />
                <span>
                  Password
                </span>
                <div
                  class="end-adornment"
                />
              </label>
            </div>
            <p
              style="margin-left: 6px; color: red;"
            >
              Password is required
            </p>
          </div>
          <div
            style="margin-left: 3px;"
          >
            <label
              class="checkbox-label"
            >
              <input
                class="checkbox-input"
                name="show-password"
                type="checkbox"
              />
              <span
                class="checkbox-text-span"
              >
                Show password
              </span>
            </label>
          </div>
        </div>
        <button
          class="button  button--default button--full-width undefined"
          disabled=""
          type="button"
        >
           
          Log In
        </button>
        <p
          style="margin-left: 6px; color: green;"
        />
      </div>
    </main>
  </div>
</div>
`;
